core_backend/app/question_answer/routers.py

```diff
@@ -91,6 +95,93 @@
 )
 
 
+@with_new_session
+async def _chat(
+    *,
+    asession: AsyncSession | None = None,
+    request: Request,
+    reset_chat_history: bool,
+    user_query: QueryBase,
+    workspace_db: WorkspaceDB,
+) -> QueryResponse | JSONResponse | dict[str, Any]:
+    """Internal chat function to process chat requests either in the background or
+    directly via the `/chat` endpoint.
+
+    The process is as follows:
+
+    1. Initialize the user query and chat histories.
+    2. Call the `search` function to get the response.
+    3. If the user query contains a WhatsApp ID, send the response via Turn.io and
+         return the payload dictionary.
+    4. If not a WhatsApp query, return the response directly.
+
+    Parameters
+    ----------
+    asession
+        The SQLAlchemy async session to use for all database connections. The default
+        for this should be `None` since it is provided by the `with_new_session`
+        decorator.
+    request
+        The FastAPI request object.
+    reset_chat_history
+        Specifies whether to reset the chat history.
+    user_query
+        The user query object.
+    workspace_db
+        The authenticated workspace object.
+
+    Returns
+    -------
+    QueryResponse | JSONResponse | dict[str, Any]
+        The query response object, an appropriate JSON response, or a dictionary
+        (for WhatsApp responses).
+    """
+
+    try:
+        # 1.
+        user_query = await init_user_query_and_chat_histories(
```

Comment:
Initializing user query and chat histories is the same for both async and background task logic.